great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -955,13 +955,40 @@ def get_validator(
                     )
                 )
 
-        return self.get_validator_using_batch_list(
+        self._check_len_batch_list(batch_list)
+        # For ZEP datasources, we are trying to simplify our datatypes and we don't currently have `batch_definitions`
+        # defined on a `Batch`. Because of this, we grab the execution engine using a batch_request object.
+        # Once a ZEP `Batch` gets defined we will clean this up and with a more appropriate type check.
+        # NOTE: for non-ZEP workflows, both branches are equivalent.
+        # NOTE: `batch_request_list != [None]` is a workaround when no batch, batch_list, batch_request,
+        # batch_request_list is passed in. This should not be allowed but many tests currently do this and need to be
+        # fixed.
+        execution_engine: ExecutionEngine
+        if batch_request_list and batch_request_list != [None] and datasource_name:
```

Comment:
The expectation calculation pathway will go down this first branch which it what the ZEP workflow will do. This avoids looking at a `batch.batch_definition` to figure out the `execution_engine`. Both pathways generate the same execution engine, they just look for the datasource in different places.